examples/nextjs-todo/components/ui/Todo.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useEffect } from "react";
+
+import useClient from "@/arke/useClient";
+import { TBaseParameter, TUnit } from "@arkejs/client";
+import { Form, FormField } from "@arkejs/form";
+import { Dispatch, SetStateAction } from "react";
+
+import { Button } from "@arkejs/ui";
+
+interface ITodoProps {
+  title: string | unknown;
+  description: string | unknown;
+  id: string;
+  done: boolean | unknown;
+  onSubmit: () => void;
+  onClose: (value: boolean) => void;
+  setActiveTodoId?: Dispatch<SetStateAction<string | undefined>>;
+  activeTodoId?: string;
+  isOpen?: boolean;
+}
+
+export default function Todo({
+  title,
+  description,
+  id,
+  done,
+  onSubmit,
+  onClose,
+  setActiveTodoId,
+}: ITodoProps) {
+  const client = useClient();
+
+  const [fields, setFields] = useState<TBaseParameter[]>();
+
+  useEffect(() => {
+    client.unit
+      .struct("todo", id)
+      .then((res) => setFields(res.data.content.parameters));
+  }, []);
+
+  const handleDelete = async () => {
+    client.unit
+      .delete("todo", id)
+      .then((res) => onSubmit())
+      .catch((e) => console.log("something went wrong: ", e));
+  };
+
+  const handleCheck = async (data: TUnit) => {
+    client.unit
+      .edit("todo", id, data)
+      .then((res) => onSubmit())
+      .catch((e) => console.log("something went wrong: ", e));
+  };
+
+  return (
+    <div
+      className={`relative p-4 border border-black border-b-4 border-l-4 rounded-bl-xl rounded-br-lg rounded-tr-md rounded-tl-lg w-60 min-h-[250px] ${
+        done ? "bg-slate-200" : "bg-white"
+      }`}
+    >
+      {fields && (
+        <Form
+          fields={fields}
+          onSubmit={(values: TUnit) => handleCheck(values)}
+          components={{
+            string: (props) => <input {...props} type="text" />,
+          }}
+        >
+          {({ fields }) => (
+            <div>
+              <FormField
+                id="done"
+                render={(props) => (
+                  <Button
+                    {...props}
+                    onClick={(e) => {
+                      props.onChange(!props.value);
+                      onSubmit();
+                    }}
+                    className="absolute border border-black p-0 w-6 h-6 rounded-full -top-2 -right-2 bg-white"
+                  >
+                    {done ? (
+                      <div className="bg-black w-3 h-3 rounded-full" />
+                    ) : null}
+                  </Button>
```

Comment:
Since you want to handle just `done` we can avoid to call the struct and use fields here